src/Microsoft.ML/Runtime/Internal/Tools/CSharpApiGenerator.cs

```diff
@@ -495,16 +495,58 @@ private void GenerateFooter(IndentingTextWriter writer)
             writer.WriteLine();
         }
 
-        static string GetSymbolFromType(Dictionary<string, string> typesSymbolTable, string fullTypeName, string currentNamespace)
+        /// <summary>
+        /// This methods creates a unique name for a class/struct/enum, given a type and a namespace.
+        /// It generates the name based on the <see cref="Type.FullName"/> property of the type
+        /// (see description here https://msdn.microsoft.com/en-us/library/system.type.fullname(v=vs.110).aspx).
+        /// </summary>
+        /// <param name="typesSymbolTable">A dictionary containing the names of the classes already generated.
+        /// This parameter is only used to ensure that the newly generated name is unique.</param>
```

Comment:
It is not just used to ensure the newly generated name is unique but it contains the namespace to be used for a given type across the file. The goal of having a symbol table was to not have duplicate definition of the same types by nesting them in classes; this dictionary servers as a look up table whenever we want to reference a type.